### ISSUE TITLE
Channel factory builder cannot dynamically create channels with system defined channel types

### ISSUE THREAD

**Issue by andrewfg** [openhab/openhab-core#3579]
The OH core channel builder cannot create dynamic channels based on system defined channel types. Reason is that the `.withType()` method takes a ChannelTypeUID parameter, the system defined channel types (e.g. `system.color`) contains dot `.` characters, which are not allowed in the ID part of a ChannelTypeUID.

**Comment by andrewfg** [comment]
To be more specific it fails when the ChannelTypeUID() is created from a SINGLE string argument as below..

```
Channel channel = ChannelBuilder.create(new ChannelUID(new ThingUID(new ThingTypeUID("aaa", "bbb"), "ccc"), "ddd")) 
   .withType(new ChannelTypeUID("system.color-temperature")).build();
```

I am wondering if it might work if the ChannelTypeUID() is created with TWO string arguments as follows? I am not sure if this does actually work. => Is the OH Core able to identify `system` as a valid (internal) binding ID for channel types?

```
Channel channel = ChannelBuilder.create(new ChannelUID(new ThingUID(new ThingTypeUID("aaa", "bbb"), "ccc"), "ddd"))   
    .withType(new ChannelTypeUID("system", "color-temperature")).build();
```

**Comment by andrewfg** [comment]
> Is the OH Core able to identify system as a valid (internal) binding ID for channel types?

To answer my own question; yes! 

```
public class DefaultSystemChannelTypeProvider implements ChannelTypeProvider {

    static final String BINDING_ID = "system";
```

So the real bug is simply that `new ChannelTypeUID("system.color-temperature")` should function as new `ChannelTypeUID("system", "color-temperature")`

**Comment by J-N-K** [comment]
The question is why there is a dot in the system-channeltypes. In the `DefaultSystemChannelTypeProvider` they are created like

```
public static final ChannelTypeUID SYSTEM_CHANNEL_TYPE_UID_SIGNAL_STRENGTH = new ChannelTypeUID(BINDING_ID,
            "signal-strength");
```

**Comment by andrewfg** [comment]
> The question is why there is a dot in the system-channeltypes

Indeed. The [documentation](https://www.openhab.org/docs/developer/bindings/thing-xml.html#state-channel-types) says..

```
A channel type .. can .. be referenced using a system. prefix in a channel typeId attribute in any binding
```

**Comment by J-N-K** [comment]
Digging a bit in the code I found that the `ChannelXMLResult` (which is an intermediate step in reading the XML thing definitions uses `XmlUtil.getSystemUID` to get a string version of the channel type UID and there we find:

```
    /**
     * Returns a UID in the format of {1}:{2}, where {1} is {@link #SYSTEM_NAMESPACE} and {2} is the
     * given typeId stripped of the prefix {@link #SYSTEM_NAMESPACE_PREFIX} if it exists.
     *
     * @param typeId
     * @return system uid (e.g. "system:test")
     */
    public static String getSystemUID(String typeId) {
        String type;
        if (typeId.startsWith(SYSTEM_NAMESPACE_PREFIX)) {
            type = typeId.substring(SYSTEM_NAMESPACE_PREFIX.length());
        } else {
            type = typeId;
        }
        return
```

So essentially when reading the XML, `system.foochannel` is converted to `system:foochannel`. In my opinion this should be changed. It's unexpected that we internally use `system:` and externally `system.`. @openhab/core-maintainers @openhab/add-ons-maintainers WDYT? I believe this can be easily refactored by "search & replace" in addons, but is far more consistent.

**Comment by kaikreuzer** [comment]
Just to be clear what you suggest: You want to replace `system.foochannel` by `system:foochannel`, i.e. making it the same syntax as with channel types from bindings, so that "system" becomes a reserved binding id, is this right?

**Comment by J-N-K** [comment]
Exactly. We need to figure out how to allow that in the XSD because currently the `:` is not allowed in the type id for channel definitions (which is probably why the `system.` convention was introduced).
